apps/desktop-wallet/src/storage/addresses/addressesSelectors.ts

```diff
@@ -85,15 +82,34 @@ export const makeSelectAddressesAlphAsset = () =>
 
 export const makeSelectAddressesTokens = () =>
   createSelector(
-    [selectAllFungibleTokens, selectAllNFTs, makeSelectAddressesAlphAsset(), makeSelectAddresses(), selectAllPrices],
-    (fungibleTokens, nfts, alphAsset, addresses, tokenPrices): Asset[] => {
-      const tokenBalances = getAddressesTokenBalances(addresses)
-      const tokens = calculateAssetsData([alphAsset, ...tokenBalances], fungibleTokens, nfts, tokenPrices)
-
-      return sortAssets(tokens)
-    }
+    [selectAllFungibleTokens, selectAllNFTs, makeSelectAddressesAlphAsset(), makeSelectAddresses()],
+    (fungibleTokens, nfts, alphAsset, addresses): Asset[] =>
+      calculateAssetsData([alphAsset, ...getAddressesTokenBalances(addresses)], fungibleTokens, nfts)
   )
 
+// TODO: Temp copied from shared to remove "worth" without affecting mobile wallet
+const calculateAssetsData = (tokenBalances: TokenDisplayBalances[], fungibleTokens: FungibleToken[], nfts: NFT[]) =>
+  tokenBalances.reduce((acc, token) => {
+    const fungibleToken = fungibleTokens.find((t) => t.id === token.id)
+    const nftInfo = nfts.find((nft) => nft.id === token.id)
+    const decimals = fungibleToken?.decimals ?? 0
+    const balance = BigInt(token.balance.toString())
+
+    acc.push({
+      id: token.id,
+      balance,
+      lockedBalance: BigInt(token.lockedBalance.toString()),
+      name: fungibleToken?.name ?? nftInfo?.name,
+      symbol: fungibleToken?.symbol,
+      description: fungibleToken?.description ?? nftInfo?.description,
+      logoURI: fungibleToken?.logoURI ?? nftInfo?.image,
+      decimals,
+      verified: fungibleToken?.verified
```

Comment:
Linked to our discussion: the type guards allow us to get rid of this additional field. That's cool, because we can use the unaltered data types we get from the backend.

Reply:
awesome